### ISSUE TITLE
Rough UDF Mesh

### ISSUE THREAD

**Issue by wishinghao** [xxlong0/NeuralUDF#11]
Hello, we ran 320 in DeepFashion3D and found that the extracted UDF mesh is relatively coarse. Is this a reasonable result, or did I miss any steps? The resolution is 256.

![Lark20231011-155831](https://github.com/xxlong0/NeuralUDF/assets/35792366/fba3f731-acc0-4d3b-a030-11a460dfbfbd)

**Comment by flamehaze1115** [comment]
This is because the UDF gradients near 0 level set are unstable. MeshUDF requires accurate gradients to extract meshes. You can modify the weight of the geometric regularization a bit, which can solve the problem.

**Comment by wishinghao** [comment]
I tried adjusting the weight of the geometric regularization term but didn't observe a significant improvement in the UDF mesh. Could you provide some debugging tips or suggest appropriate hyperparameters as an example? I'm adjusting the weight of this  term: 
`sparse_error = torch.exp(-self.sparse_scale_factor * udf).sum(dim=1, keepdim=False).mean()`

**Comment by flamehaze1115** [comment]
Could you please your weights on this object, I can help to figure out.

**Comment by wishinghao** [comment]
Thank you very much for your assistance. I initially tried the training parameters you provided, with `bash bashs/bash_garment_blending.sh --gpu 0 --case scan320 -s 0.001` and `bash bashs/bash_garment_blending_ft.sh --gpu 0 --case scan320 -s 0.01`. I also tried "scan 300," and the resulting UDF mesh is still coarse. 
![image](https://github.com/xxlong0/NeuralUDF/assets/35792366/18e7d90f-633f-451e-9855-6cac87df30fb)
Then I attempted to directly increase the weight of "sparse_error" to 0.01 and 0.05 in the first stage. However, the issue still remains unresolved.
I'm using the V100 GPU, and I installed Python according to the conda_env.yml provided in the code.
